### PR TITLE
feat(data_collecting_tool): add multiple courses

### PR DESCRIPTION
**Pull request by YoshihiroKogure**
### Course Selection
Added several courses to accommodate various cases of data collection.
You can use following courses by setting COURSE_NAME to "eight_course",  "u_shaped_return", "straight_line_positive" or "straight_line_negative" in `data_collecting_trajectory_publisher.py`.
( Both "straight_line_positive" and "straight_line_negative" represent straight line courses, but the direction of travel of the course is reversed.)

- Eight course
eight

- U-shaped course
u_shaped

- Straight line course
straight

### Increased Frequency
There was a problem with the frequency dropping when plotting data counts in real time. This problem was mitigated via threading and Callback Groups.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py [hunk 1]
```diff
@@ -410,22 +749,52 @@ def updateNominalTargetTrajectory(self):
 
         long_side_margin = 5
         long_side_margin = 5
-        total_distance = ld * (1 + np.pi) * 2
+        # total_distance = ld * (1 + np.pi) * 2
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py:752]: please remove commentout if it is not necessary.

File: control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py [hunk 2]
```diff
@@ -372,7 +711,7 @@ def updateNominalTargetTrajectory(self):
         lb = (l2 + l4) / 2
         if np.abs(la - lb) < 1e-6:
             la += 0.1  # long_side_length must not be equal to short_side_length
-        ld = np.sqrt(la**2 + lb**2)
+        # ld = np.sqrt(la**2 + lb**2)
```
Comment (kosuke55) [inline control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py:714]: please remove commentout if it is not necessary.

**Comment by kosuke55** [comment]
confirmed that it works in my environment.

image

internal slack:  https://star4.slack.com/archives/C05UHMNSE5C/p1726122855613459?thread_ts=1726120701.103249&cid=C05UHMNSE5C

**Comment by kosuke55** [comment]
> I don't know why, but there are some unnecessary differences related to m

rebased on main to fix commits

File: control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py (hunk 1 above)
Comment (YoshihiroKogure) [reply control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py]: Thank you for your feedback. I removed commentout.

File: control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py (hunk 2 above)
Comment (YoshihiroKogure) [reply control_data_collecting_tool/scripts/data_collecting_trajectory_publisher.py]: Thank you for your feedback. I removed commentout.

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**